From d7259063208e80bc1b1db98503940a0c37db9908 Mon Sep 17 00:00:00 2001
From: Fancy Fang <B47543@freescale.com>
Date: Wed, 8 Jan 2014 15:21:00 +0800
Subject: [PATCH 253/256] ENGR00294173 G2D: fix some build warnings for g2d
 library

Fix the following build warnings:
"
g2d.c: In function 'g2d_open':
g2d.c:205:2: warning: implicit declaration of function 'close' [-Wimplicit-function-declaration]
close(fd);
^
g2d.c: In function 'g2d_copy':
g2d.c:474:6: warning: unused variable 'ret' [-Wunused-variable]
int ret;
^
g2d.c: In function 'g2d_clear':
g2d.c:556:3: warning: format '%s' expects argument of type 'char *', but argument 2 has type 'struct g2d_surface *' [-Wformat=]
g2d_printf("%s: invalid clear area\n", area);
^
g2d.c:545:15: warning: unused variable 'clrColor' [-Wunused-variable]
unsigned int clrColor;
^
g2d.c:544:6: warning: unused variable 'ret' [-Wunused-variable]
int ret;
^
g2d.c: In function 'g2d_blit':
g2d.c:590:19: warning: unused variable 'vMirror' [-Wunused-variable]
int hMirror = 0, vMirror = 0;
^
g2d.c:590:6: warning: unused variable 'hMirror' [-Wunused-variable]
int hMirror = 0, vMirror = 0;
^
g2d.c:586:6: warning: unused variable 'ret' [-Wunused-variable]
int ret;
^
g2d.c:868:23: warning: 'third_param' may be used uninitialized in this function [-Wmaybe-uninitialized]
third_param->width  = out_param->width;
"

Signed-off-by: Fancy Fang <B47543@freescale.com>
---
 g2d/g2d.c | 10 +++-------
 1 file changed, 3 insertions(+), 7 deletions(-)

diff --git a/g2d/g2d.c b/g2d/g2d.c
index 48ec191..d9118ff 100644
--- a/g2d/g2d.c
+++ b/g2d/g2d.c
@@ -16,6 +16,7 @@
 #include <sys/ioctl.h>
 #include <sys/mman.h>
 #include <pthread.h>
+#include <unistd.h>
 #include <linux/pxp_device.h>
 #include "g2d.h"
 
@@ -471,7 +472,6 @@ int g2d_free(struct g2d_buf *buf)
 #define PXP_COPY_THRESHOLD (16*16*4)
 int g2d_copy(void *handle, struct g2d_buf *d, struct g2d_buf* s, int size)
 {
-	int ret;
 	unsigned int blit_size;
 	struct pxp_config_data pxp_conf;
 	struct pxp_layer_param *src_param = NULL, *out_param = NULL;
@@ -541,8 +541,6 @@ int g2d_copy(void *handle, struct g2d_buf *d, struct g2d_buf* s, int size)
 
 int g2d_clear(void *handle, struct g2d_surface *area)
 {
-	int ret;
-	unsigned int clrColor;
 	struct pxp_config_data pxp_conf;
 	struct pxp_layer_param *out_param;
 	struct g2dContext *context = (struct g2dContext *)handle;
@@ -553,7 +551,7 @@ int g2d_clear(void *handle, struct g2d_surface *area)
 	}
 
 	if (area == NULL) {
-		g2d_printf("%s: invalid clear area\n", area);
+		g2d_printf("%s: invalid clear area\n", __func__);
 		return -1;
 	}
 
@@ -583,11 +581,9 @@ int g2d_clear(void *handle, struct g2d_surface *area)
 int g2d_blit(void *handle, struct g2d_surface *src, struct g2d_surface *dst)
 {
 	int dest_bpp;
-	int ret;
 	int srcRotate, dstRotate;
 	struct pxp_config_data pxp_conf;
-	struct pxp_layer_param *src_param, *out_param, *third_param;
-	int hMirror = 0, vMirror = 0;
+	struct pxp_layer_param *src_param, *out_param, *third_param = NULL;
 	unsigned int srcWidth,srcHeight,dstWidth,dstHeight;
 	struct g2dContext *context = (struct g2dContext *)handle;
 
-- 
1.8.0

